include/exadg/structure/spatial_discretization/operator.cpp

```diff
@@ -164,6 +162,8 @@ Operator<dim, Number>::distribute_dofs()
   // call deal.II utility function to add Dirichlet constraints
   add_homogeneous_dirichlet_constraints(affine_constraints, dof_handler, map_bid_to_mask);
 
+  // compress constraints *once after* complete setup
```

Comment:
Maybe we should add a comment in line 142 that `affine_constraints_periodicity_and_hanging_nodes` must be in "un-closed" state since we later want to add additional constraints.

I think such a comment would help me to better understand the code.